packages/block-library/src/social-link/edit.js

```diff
@@ -138,16 +137,18 @@ const SocialLinkEdit = ( {
 	// Use internal state instead of a ref to make sure that the component
 	// re-renders when the popover's anchor updates.
 	const [ popoverAnchor, setPopoverAnchor ] = useState( null );
-	const isContentOnlyMode = useBlockEditingMode() === 'contentOnly';
```

Comment:
I think this is the preferred method for fetching block editing mode in blocks. It doesn't create a store subscription in this case and retrieves the value from the context. The mode setting effect also bails early when no mode argument is provided.

The proposed change will add a new `block-editor` subscription for each Social Icon block, which we're generally trying to avoid. The `block-editor` is the most active store in the project and is affected by the O(n) subscription problem.

Reply:
Oh - I looked briefly at the `useBlockEditingMode` docs and didn't notice that part. I assumed it was more for setting a mode than retrieving. Spent some time brushing back up on block context and subscription patterns. Thanks for pointing this out and explaining it!